packages/odyssey-react-mui/src/OdysseyProvider.tsx

```diff
@@ -51,32 +62,73 @@ const OdysseyProvider = <SupportedLanguages extends string>({
   stylisPlugins,
   themeOverride,
   translationOverrides,
-}: OdysseyProviderProps<SupportedLanguages>) => (
-  <OdysseyCacheProvider
-    emotionRoot={emotionRoot}
-    emotionRootElement={emotionRootElement}
-    hasShadowDom={Boolean(shadowRootElement || shadowDomElement)}
-    nonce={nonce}
-    stylisPlugins={stylisPlugins}
-  >
-    <OdysseyThemeProvider
-      designTokensOverride={designTokensOverride}
-      shadowRootElement={shadowRootElement || shadowDomElement}
-      themeOverride={themeOverride}
+}: OdysseyProviderProps<SupportedLanguages>) => {
+  const { contrastMode } = useContrastContext();
+
+  const odysseyTokens = useMemo(
+    () => ({ ...Tokens, ...designTokensOverride }),
+    [designTokensOverride],
+  );
+
+  const odysseyTheme = useMemo(
+    () =>
+      createOdysseyMuiTheme({
+        odysseyTokens,
+        shadowRootElement: shadowRootElement || shadowDomElement,
+      }),
+    [odysseyTokens, shadowDomElement, shadowRootElement],
+  );
+
+  const odysseyThemeWithBackground = useMemo(
+    () =>
+      createTheme({
+        ...odysseyTheme,
+        odysseyContrastMode: contrastMode,
+      }),
+    [odysseyTheme, contrastMode],
+  );
+
+  const customOdysseyTheme = useMemo(
+    () =>
+      themeOverride
+        ? createTheme(deepmerge(odysseyThemeWithBackground, themeOverride))
+        : odysseyThemeWithBackground,
+    [odysseyThemeWithBackground, themeOverride],
+  );
+
+  return (
+    <OdysseyCacheProvider
+      emotionRoot={emotionRoot}
+      emotionRootElement={emotionRootElement}
+      hasShadowDom={Boolean(shadowRootElement || shadowDomElement)}
+      nonce={nonce}
+      stylisPlugins={stylisPlugins}
     >
-      {/* This component creates a div; for flexibility of layout of children, make it inherit its parent's height */}
-      <ScopedCssBaseline sx={scopedCssBaselineStyles}>
-        <OdysseyTranslationProvider<SupportedLanguages>
-          languageCode={languageCode}
-          translationOverrides={translationOverrides}
-        >
-          {children}
-        </OdysseyTranslationProvider>
-      </ScopedCssBaseline>
-    </OdysseyThemeProvider>
```

Comment:
@benlister-okta I'm not sure we can remove the `OdysseyThemeProvider` here to be honest

Reply:
I guess you are passing the tokens along to `OdysseyDesignTokensContext` here, so, maybe we can

Reply:
I'm not sure what you mean by this.^ We are not using `OdysseyThemeProvider` at all now

Reply:
I meant it was implementing the same functionality. I see what you mean though, it no longer uses OdysseyThemeProvider at all so it would be weird and confusing to have the functionality duplicated.

I just updated to revert back to using OdysseyThemeProvider and extend it with the contrast stuff